cli/pkg/kctrl/cmd/core/kubeconfig_flags.go

```diff
@@ -31,6 +31,11 @@ func (f *KubeconfigFlags) Set(cmd *cobra.Command, flagsFactory FlagsFactory, opt
 	kubeconfigYamlEnvVariableKey := fmt.Sprintf("%s_KUBECONFIG_YAML", strings.ToUpper(opts.BinaryName))
 	f.YAML = NewKubeconfigYAMLFlag(kubeconfigYamlEnvVariableKey)
 	cmd.PersistentFlags().Var(f.YAML, "kubeconfig-yaml", fmt.Sprintf("Kubeconfig contents as YAML ($%s)", kubeconfigYamlEnvVariableKey))
+
+	if opts.DefaultKubeconfigOverrideContext != "" && opts.DefaultKubeconfigOverridePath != "" {
```

Comment:
Is it possible that someone might set one of these and not both? (Based on the individual flag name, they shouldn't depend on each other)

Reply:
The expectation is that if you are consuming these they are both submitted. For the use case these are supposed to address, it is reasonable to expect both at all times.

Reply:
It might make sense to separate these none the less, even we do not have a case for it now

Reply:
I think if we want to handle precedence order of 1st --kubeconfig, then KUBECONFIG and then any other (eg. tanzu cli context), then we need to handle the flags differently here.  we will need to check if nothing else is set, then we set the defaults.

Reply:
Made this change, do take a peek! I will test out the precedence as well. We should account for this in our tests